### ISSUE TITLE
Enhance clustering on downstream task

### ISSUE THREAD

**Issue by chris-tkinter** [UKPLab/sentence-transformers#1101]
Hi all! Thanks for making this awesome project!
I have a question, so I would like to use sentence transformers for a clustering task in my own dataset, and my goal is to show that similar sentences (sentences from the same class) would belong to the same cluster. 

I have been using model `paraphrase-distilroberta-base-v2` in a zero-shot setting: I just use the weight from the pre-trained model and plot the clustering. However, I found that this clustering is not as strong as a fine-tuned BERT model, which makes sense since the sentence transformer model has not been fine-tuned on the downstream task yet. But I am wondering how should I fine tune to improve the clustering performance?

Since I do not have the score between pairs of the sentences in my dataset, I think I couldn't use the same training process in the sentence transformer model; however, I do have labels for each of the sentences.

I was thinking of 2 approaches:

1. fine tune the underlying transformer (distilroberta-base) in this case

2. structure my dataset to (anchor, positive) pairs, where positive comes from sentences of the same class, and train MultipleNegativesRankingLoss, as mentioned in #1022

Since my computation power is limited it might take me a long time for either approach, therefore I just asked my question here before trying both. Thanks!!

**Comment by ScottishFold007** [comment]
> Hi all! Thanks for making this awesome project!
> I have a question, so I would like to use sentence transformers for a clustering task in my own dataset, and my goal is to show that similar sentences (sentences from the same class) would belong to the same cluster.
> 
> I have been using model `paraphrase-distilroberta-base-v2` in a zero-shot setting: I just use the weight from the pre-trained model and plot the clustering. However, I found that this clustering is not as strong as a fine-tuned BERT model, which makes sense since the sentence transformer model has not been fine-tuned on the downstream task yet. But I am wondering how should I fine tune to improve the clustering performance?
> 
> Since I do not have the score between pairs of the sentences in my dataset, I think I couldn't use the same training process in the sentence transformer model; however, I do have labels for each of the sentences.
> 
> I was thinking of 2 approaches:
> 
> 1. fine tune the underlying transformer (distilroberta-base) in this case
> 2. structure my dataset to (anchor, positive) pairs, where positive comes from sentences of the same class, and train MultipleNegativesRankingLoss, as mentioned in [Clustering for domain specific content #1022](https://github.com/UKPLab/sentence-transformers/issues/1022)
> 
> Since my computation power is limited it might take me a long time for either approach, therefore I just asked my question here before trying both. Thanks!!

You can try simcse to get a better semantic similarity representation even without annotated corpus; in addition, you can try colab if your computing power is not enough

**Comment by nreimers** [comment]
@ScottishFold007 SimCSE  does not work well for learning domain specific language, please don't use it. It is worse than (strong) out-of-the-box pre-trained models like the paraphrase v2 models. We have analyzed it here: https://sbert.net/papers/TSDAE.pdf

Training with MultipleNegativesRankingLoss on your own data makes sense.

**Comment by AlexYoung757** [comment]
@nreimers  hello the link : https://sbert.net/papers/TSDAE.pdf 。it's not available

**Comment by nreimers** [comment]
@AlexYoung757 You can find the link here: https://arxiv.org/abs/2104.06979
